snapcraft/cli/remote.py

```diff
@@ -224,9 +228,7 @@ def _list_architectures(project, arch: str) -> List[str]:
     arch_from_yaml = _project_architectures(project)
     if arch_from_yaml:
         if arch:
-            echo.warning(
-                "Architecture list already set in snapcraft.yaml, ignoring --arch option."
-            )
+            raise errors.ConflictingArchListError()
```

Comment:
this could be click.BadOptionUsage instead of a full blown exception in errors (given you are doing the check here and not internally in the remote_build package

Reply:
It makes sense. Fixed this along with the code refactoring describe three comments below.